crates/bevy_reflect/src/lib.rs

```diff
@@ -620,6 +620,10 @@ pub mod __macro_exports {
     ///
     /// This trait has a blanket implementation for all types that implement `GetTypeRegistration`
     /// and manual implementations for all dynamic types (which simply do nothing).
+    #[diagnostic::on_unimplemented(
+        message = "`{Self}` does not implement `GetTypeRegistration` so cannot be registered for reflection",
+        note = "consider annotating `{Self}` with `#[derive(Reflect)]`"
+    )]
     pub trait RegisterForReflection {
```

Comment:
Is it intended for this to mention `GetTypeRegistration` instead of `RegisterForReflection`? Is it also intended that the message differs from the one for `GetTypeRegistration`?